### PR TITLE
Pie 1169/attendance read view

### PR DESCRIPTION
**Pull request by cjhaddad**
## 🍂 Before You Submit

* [x] Did you write & run tests and linters?
* [ ] Did you run Google Lighthouse and/or WebAIM (Wave) on UI components in your PR?
* [x] Does your PR contain any required translations?
* [ ] Are your primary keys UUIDs on any new tables?

## 🛷 Deployment

* [ ] Migrations
* [ ] Dependencies

### CODE REVIEW

**Comment by cjhaddad** [comment]
@katelovescode taking the WIP off of this. however, cypress not added. wondering if you'd want to pair with me on that since I haven't written before. would likely go faster and I can ask questions easily.

**Comment by cjhaddad** [comment]
@katelovescode  the failing be tests have to do with the same issues we've been seeing, correct?

File: spec/cypress/integration/attendanceView.spec.js [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+  })
+
+  describe('inputAttendance', () => {
+    it('directs you to attendance edit button', () => {
+      cy.get('[data-cy=inputAttendance]').click()
+      cy.location('pathname').should('eq', '/attendance/edit')
+    })
+  })
+
+  describe('weekPicker', () => {
```
Comment (katelovescode) [inline spec/cypress/integration/attendanceView.spec.js:93]: 😍

File: client/src/i18n/locales/en/index.json [hunk 2]
```diff
@@ -221,5 +222,20 @@
   "addAnotherDate": "Add another date",
   "success": "Success!",
   "successText": "You just entered some attendance history for the kids in your care.",
-  "gotToDashboard": "Go to dashboard"
+  "gotToDashboard": "Go to dashboard",
+  "dateError": "Please select a date",
+
+  "inputAttendance": "Input Attendance",
+  "name": "Name",
+  "mon": "on",
```
Comment (katelovescode) [inline client/src/i18n/locales/en/index.json:230]:
```suggestion
  "mon": "mon",
```

File: client/src/i18n/locales/es/index.json [hunk 3]
```diff
@@ -221,5 +222,20 @@
   "addAnotherDate": "Otra fecha",
   "success": "¡Registro exitoso!",
   "successText": "Acabas de registrar el historial de asistencia para los niños bajo tu supervisión.",
-  "gotToDashboard": "Ir al panel principal"
+  "gotToDashboard": "Ir al panel principal",
+  "dateError": "Por favor selecciona una fecha",
+
+  "inputAttendance": "Ingresar asistencia",
+  "name": "Nombre",
+  "mon": "mon",
```
Comment (katelovescode) [inline client/src/i18n/locales/es/index.json:230]: Should these abbreviations be the Spanish names of the months?

File: spec/cypress/integration/attendanceView.spec.js [hunk 4]
```diff
@@ -0,0 +1,107 @@
...
+      cy.get('[data-cy=backWeekButton]').click()
+      cy.contains(weekPickerText(dayjs().weekday(-7)))
+    })
+
+    it('allows you to view next week attendance', () => {
+      cy.get('[data-cy=forwardWeekButton]').click()
+      cy.contains(weekPickerText(dayjs().weekday(7)))
+    })
+  })
+})
```
Comment (katelovescode) [inline spec/cypress/integration/attendanceView.spec.js:107]: Can we add some screen size expectations?  I don't know if Cypress handles that but I'd love to see an expectation that says "if the screen size is too small, cy.contains(the text that tells you to go to desktop)"

File: spec/cypress/integration/attendanceView.spec.js [hunk 5]
```diff
@@ -0,0 +1,107 @@
...
+      url: '/attendances'
+    }).as('attendances')
+    cy.visit('/attendance')
+  })
+
+  describe('content', () => {
+    it('renders content', () => {
+      cy.contains(childFullName)
+      cy.contains('4 hrs 0 mins')
+      cy.contains('no info')
```
Comment (katelovescode) [inline spec/cypress/integration/attendanceView.spec.js:82]: Can you indicate how many times you expect to see 'no info'?  My guess is there will be one day with an attendance and six days with no info, can you do (pseudocode): `cy.contains('no info').exactly(6).times` or something like that?  Not sure if Cypress has this.  Maybe also expect the inputAttendance button to be rendered?

File: spec/cypress/integration/attendanceView.spec.js [hunk 6]
```diff
@@ -0,0 +1,107 @@
...
+const firstName = name.firstName()
+const fullName = name.findName(firstName)
+const email = internet.email(firstName)
+const password = internet.password()
+const checkIn = new Date(Date.now() - 21600000)
+const checkOut = new Date(Date.now() - 7200000)
+let childFullName
+
+const weekPickerText = (date = dayjs()) =>
+  `${date.weekday(0).format('MMM D')} - ${date.weekday(6).format('MMM D')}`
```
Comment (katelovescode) [inline spec/cypress/integration/attendanceView.spec.js:18]: 😍

File: spec/cypress/integration/attendanceView.spec.js [hunk 7]
```diff
@@ -0,0 +1,107 @@
...
+import weekday from 'dayjs/plugin/weekday'
+
+dayjs.extend(weekday)
+
+const { name, internet } = faker
+const firstName = name.firstName()
+const fullName = name.findName(firstName)
+const email = internet.email(firstName)
+const password = internet.password()
+const checkIn = new Date(Date.now() - 21600000)
```
Comment (katelovescode) [inline spec/cypress/integration/attendanceView.spec.js:13]: Can we just comment to document how much time this is subtracting in human-readable terms?  i.e.
```
// Date now - 6 hours
```
or whatever it is

File: client/src/i18n/locales/es/index.json [hunk 8]
```diff
@@ -221,5 +222,20 @@
...
+  "tues": "tues",
+  "wed": "wed",
+  "thurs": "thurs",
+  "fri": "fri",
+  "sat": "sat",
+  "sun": "sun",
+
+  "screenSize": "Tamaño de pantalla no compatible",
+  "incompatibleMsg": "O tu navegador is demasiado pequeño o estás en un aparato movil. Por favor cambia a una computadora o tableta para ver esta pagina.",
+  "noInfo": "no info"
```
Comment (katelovescode) [inline client/src/i18n/locales/es/index.json:240]: Should this be in Spanish?

File: client/src/_shared/Header.js [hunk 9]
```diff
@@ -93,13 +109,41 @@ export function Header() {
...
+              </Button>
+            </div>
+          </div>
+        </div>
+      ) : (
+        <div className={`text-2xl font-semibold flex-grow`}>
+          Pie for Providers
+        </div>
+      )}
+      {/* <div
```
Comment (katelovescode) [inline client/src/_shared/Header.js:140]: Should this be removed?

File: client/src/AttendanceView/WeekPicker.js [hunk 10]
```diff
@@ -0,0 +1,39 @@
+import React from 'react'
+import PropTypes from 'prop-types'
+import { Button } from 'antd'
+import { LeftOutlined, RightOutlined } from '@ant-design/icons'
+
+export function WeekPicker({ dateSelected, handleDateChange }) {
```
Comment (katelovescode) [inline client/src/AttendanceView/WeekPicker.js:6]: I love this compact little component!

**Review by katelovescode — COMMENTED** [review]
Looks really great, runs perfectly!  Just a couple of code hygiene comments and a few more spec requests. ❤️

File: client/src/i18n/locales/es/index.json (hunk 8 above)
Comment (cjhaddad) [reply client/src/i18n/locales/es/index.json:240]: "no info" is also the spanish translation. confirmed this with Chelsea

File: client/src/AttendanceView/WeekPicker.js (hunk 10 above)
Comment (cjhaddad) [reply client/src/AttendanceView/WeekPicker.js:6]: 💥

File: client/src/i18n/locales/es/index.json (hunk 3 above)
Comment (cjhaddad) [reply client/src/i18n/locales/es/index.json:230]: yes!

File: spec/cypress/integration/attendanceView.spec.js (hunk 5 above)
Comment (cjhaddad) [reply spec/cypress/integration/attendanceView.spec.js:82]: yes!

File: spec/cypress/integration/attendanceView.spec.js (hunk 7 above)
Comment (cjhaddad) [reply spec/cypress/integration/attendanceView.spec.js:13]: yes!

**Review by katelovescode — APPROVED** [review]
Excellent work, @cjhaddad !  Thanks so much!

**Comment by cjhaddad** [comment]
> Excellent work, @cjhaddad ! Thanks so much!

of course! should I merge?

**Pull request merged**